src/components/LiveCursors/Cursors.tsx

```diff
@@ -0,0 +1,119 @@
+import { Space, SpaceMember } from '@ably-labs/spaces'
+import CursorSvg from './CursorSvg'
+import useCursor from './use-cursor'
+
+// 💡 This component is used to render the cursor of the user
+const YourCursor = ({
+  user,
+  space,
+}: {
+  user: SpaceMember | undefined
+  space: Space | undefined
+}) => {
+  const [cursorPosition, setCursorPosition] = useState<{
+    [key: string]: number
+  }>({ left: 100, top: 100 })
+
+  const handleSelfCursorMove = useCursor(setCursorPosition, space)
+
+  return (
+    <div
+      className="absolute"
+      onMouseMove={(e) => handleSelfCursorMove(e)}
+      style={{
+        top: `${cursorPosition.top + 30}px`,
+        left: `${cursorPosition.left}px`,
+      }}
+    >
+      <CursorSvg
+        startColor={user?.profileData.userColors.split(' ')[2]}
+        endColor={user?.profileData.userColors.split(' ')[3]}
+        id={user?.connectionId}
+      />
+      <div
+        className={`px-4 py-2 m-2 bg-gradient-to-r ${
+          user?.profileData.userColors.split(' ')[0]
+        } ${
+          user?.profileData.userColors.split(' ')[1]
+        } rounded-full text-sm text-white`}
+      >
+        {user?.profileData.name} (You)
+      </div>
+    </div>
+  )
+}
+
+// 💡 This component is used to render the cursors of other users in the space
+const MemberCursors = ({
+  otherUsers,
+  space,
+  selfConnectionId,
+}: {
+  otherUsers: SpaceMember[]
+  space?: Space
+  selfConnectionId?: string
+}) => {
+  const [positions, setPositions] = useState<{
+    [connectionId: string]: { x: number; y: number }
+  }>({})
+
+  useEffect(() => {
+    if (!space) return
+
+    const pointer = space.cursors.get('space-pointer')
+
+    pointer.on(
+      'cursorUpdate',
+      (event: { connectionId: string; position: { x: number; y: number } }) => {
+        // 💡 Ignore our own cursor
+        if (event.connectionId === selfConnectionId) return
+
+        setPositions((positions) => ({
+          ...positions,
+          [event.connectionId]: event.position,
+        }))
+      }
+    )
+    return () => {
+      pointer.off()
+    }
+  }, [space])
+
+  return (
+    <>
+      {otherUsers.map(({ connectionId, profileData }) => {
+        if (!positions[connectionId]) return
+
+        return (
+          <div
+            key={connectionId}
+            id={`member-cursor-${connectionId}`}
+            className="absolute"
+            style={{
+              left: `${positions[connectionId].x}px`,
+              top: `${positions[connectionId].y}px`,
+            }}
+          >
+            <CursorSvg
+              startColor={profileData.userColors.split(' ')[2]}
+              endColor={profileData.userColors.split(' ')[3]}
+              id={connectionId}
+            />
+            <div
+              className={`px-4 py-2 m-2 bg-gradient-to-r ${
+                profileData.userColors.split(' ')[0]
+              } ${
+                profileData.userColors.split(' ')[1]
```

Comment:
One thing that you could do here is destructure `userColors` to give them names:

```ts
const [gradientStartClass, gradientEndClass, startColor, endColor] = profileData.userColors;
```

(my names are just suggestions here, I don't really know what 0 and 1 are here)

Reply:
Or, if we're using these colours just, maybe it's worth _not using_ a string for it but an object with key names? that would give us a better opportunity to describe the data structure we're handling:

```js
profileData.colors.twGradientStartClass
```